sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -394,143 +577,141 @@ static rom_error_t rom_ext_attestation_silicon(void) {
   // TODO(cfrantz): setup sealing binding to value specified in owner
   // configuration block.
-  HARDENED_RETURN_IF_ERROR(
-      sc_keymgr_owner_advance(/*sealing_binding=*/&zero_binding_value,
-                              /*attest_binding=*/&boot_measurements.bl0,
-                              owner_manifest->max_key_version));
+  HARDENED_RETURN_IF_ERROR(sc_keymgr_owner_advance(
+      /*sealing_binding=*/&zero_binding_value,
+      /*attest_binding=*/attest_binding, owner_manifest->max_key_version));
   HARDENED_RETURN_IF_ERROR(otbn_boot_cert_ecc_p256_keygen(
-      kDiceKeyCdi1, &cdi_1_pubkey_id, &curr_attestation_pubkey));
-  hardened_bool_t cert_valid = kHardenedBoolFalse;
-  HARDENED_RETURN_IF_ERROR(cert_x509_asn1_check_serial_number(
-      dice_cert_obj.cert_body_p, /*offset=*/0,
-      (uint8_t *)cdi_1_pubkey_id.digest, &cert_valid, /*out_cert_size=*/NULL));
-  if (launder32(cert_valid) == kHardenedBoolFalse) {
-    HARDENED_CHECK_EQ(cert_valid, kHardenedBoolFalse);
+      kDiceKeyCdi1, &dice_chain.subject_pubkey_id, &dice_chain.subject_pubkey));
+
+  // Check if the current CDI_0 cert is valid.
+  HARDENED_RETURN_IF_ERROR(dice_chain_load_cert_obj("CDI_1", 6));
+  if (launder32(dice_chain.cert_valid) == kHardenedBoolFalse) {
+    HARDENED_CHECK_EQ(dice_chain.cert_valid, kHardenedBoolFalse);
     dbg_printf("CDI_1 certificate not valid. Updating it ...\r\n");
-    uint32_t updated_cert_size = kCdi1MaxCertSizeBytes;
-    // TODO(#19596): add owner configuration block measurement to CDI_1 cert.
-    HARDENED_RETURN_IF_ERROR(dice_cdi_1_cert_build(
-        (hmac_digest_t *)boot_measurements.bl0.data,
-        (hmac_digest_t *)zero_binding_value.data,
-        owner_manifest->security_version, &cdi_1_key_ids,
-        &curr_attestation_pubkey, cdi_1_cert, &updated_cert_size));
     // Update the cert page buffer.
-    size_t cert_page_left = dice_certs_buffer_space_remaining();
-    HARDENED_RETURN_IF_ERROR(perso_tlv_cert_obj_build(
-        "CDI_1", /*needs_endorsement=*/false, cdi_1_cert, updated_cert_size,
-        dice_cert_obj.obj_p, &cert_page_left));
-    dice_certs_page_dirty = kHardenedBoolTrue;
-    // Reload the cert perso LTV object.
-    HARDENED_RETURN_IF_ERROR(perso_tlv_get_cert_obj(
-        dice_cert_obj.obj_p, dice_certs_buffer_space_remaining(),
-        &dice_cert_obj));
-    HARDENED_RETURN_IF_ERROR(rom_ext_attestation_increment_cert_offset());
-    // If the CDI_1 cert is updated, it could be smaller than the previous CDI_1
-    // cert, so we make sure to clear out the rest of the buffer.
-    memset(dice_cert_obj.obj_p, 0, dice_certs_buffer_space_remaining());
+    size_t updated_cert_size = kScratchCertSizeBytes;
+    // TODO(#19596): add owner configuration block measurement to CDI_1 cert.
+    HARDENED_RETURN_IF_ERROR(
+        dice_cdi_1_cert_build((hmac_digest_t *)attest_binding->data,
+                              (hmac_digest_t *)zero_binding_value.data,
```

Comment:
The first parameter should be the BL0 (owner fw) measure as it was before: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/cert/dice.c;l=208?q=dice.c&ss=opentitan%2Fopentitan
The DiceTcbInfo extension of the cert contains both measurements that were combined to create the attest_binding measurement.

The second parameter can be 0 for now until we cherry pick this to `earlgrey_1.0.0` branch. On said branch the second parameter is the owner configuration block measurement, since that branch has the most up to date ROM_EXT with ownership transfer (https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/cert/dice.c;l=208?q=dice.c&ss=opentitan%2Fopentitan)

Reply:
Sure, renamed them to make it more clear.
The parameters are the same as before.